07-13-16-language-plugins.md

```diff
@@ -0,0 +1,97 @@
+## A Milestone for Tessel
+This month we merged one of the most significant pull requests for the Tessel project: [adding Rust deployment support](https://github.com/tessel/t2-cli/pull/790). For the past three years we've focused exclusively on deploying JavaScript on microcontrollers. This is our first step investing in a new language, a new ecosystem, and a new community. However, the purpose of this post is not to dive into the benefits of Rust, but rather to showcase the ease of adding new language support to the Tessel CLI.
+
+Despite the significance of this addition to the CLI codebase, the amount of code we modified was relatively small. We simply added two main files (along with minor modifications to a handful of other tests): [one with the implementation](https://github.com/tessel/t2-cli/blob/b8cd1f34398c2d188fa188a216b4b9b4dd206dc4/lib/tessel/deployment/rust.js) and one with [unit tests](https://github.com/tessel/t2-cli/blob/b8cd1f34398c2d188fa188a216b4b9b4dd206dc4/test/unit/deployment/rust.js). The simplicity of the feature addition is a testament [@rwaldron](https://github.com/rwaldron)'s hard work building the CLI's language plugin system. In this post, I want to demonstrate how straight forward it is to add support for a new language to the CLI using this plugin system.
+
+The process for deploying an arbitrary code bundle, regardless of language, to Tessel can be generalized. There are six functions that need to be exported for any language deployment mechanism:
```

Comment:
First sentence is messy/hard to understand. Try: "You can deploy an arbitrary code bundle to Tessel 2 (in any language) with the same six general functions:" or similar